### PR TITLE
ci(swift): release swift bindings to tag on tag creation

### PR DESCRIPTION
**Pull request by antbern**
## Motivation and Context

Again inspired by the CI in `lapce`: https://github.com/lapce/lapce/blob/master/.github/workflows/release.yml

- Run workflow when tag is pushed, take tag name as tag name of release to create in swift bindings repo.
- Allow to run workflow manually and specify a tag name.

Successful run of workflow with a manually provided tag: https://github.com/ETOSPHERES-Labs/etopay-sdk/actions/runs/14083765994/job/39443297905

Release created in swift bindings: https://github.com/ETOSPHERES-Labs/etopay-sdk-swift/releases/tag/v0.0.1

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] ~~Code follows project style and guidelines.~~
- [ ] ~~No redundant or duplicate code.~~
- [ ] ~~No added new dependencies without clear justification and approval.~~

### Documentation

- [ ] ~~Added/updated relevant documentation (e.g., README, inline comments).~~
- [ ] ~~Added a CHANGELOG entry for major changes.~~

### Testing

- [ ] ~~Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.~~
- [ ] ~~Tests pass locally.~~

### Build & CI/CD

- [x] Confirmed no CI/CD pipeline issues.
- [x] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: .github/workflows/deploy-swift.yml [hunk 1]
```diff
@@ -129,16 +149,28 @@ jobs:
...
+      # set correct parameters based on if this is nightly or not
+      - if: env.TAG_NAME == 'nightly'
+        run: |
+          echo 'SUBJECT=ETOPaySDK development build' | tee -a $GITHUB_ENV
+          echo 'PRERELEASE=--prerelease' | tee -a $GITHUB_ENV
+
+      - if: env.TAG_NAME != 'nightly'
+        run: |
+          echo 'SUBJECT=ETOPaySDK release build' | tee -a $GITHUB_ENV
+          echo 'PRERELEASE=' | tee -a $GITHUB_ENV
```
Comment (kristifidani) [inline .github/workflows/deploy-swift.yml:168]: Is this duplicated or am I missing something 👀

**Review by kristifidani — COMMENTED** [review]
Very nice 💯  

Just 1 comment

File: .github/workflows/deploy-swift.yml (hunk 1 above)
Comment (antbern) [reply .github/workflows/deploy-swift.yml:168]: They are indeed different 👀 first of all the `if` conditions are different (`==` vs `!=`) and also the values that we set (eg release vs development build, the `PRERELEASE` var is empty in the second one).

Comment (kristifidani) [reply .github/workflows/deploy-swift.yml:168]: Ahhh nvm now i noticed the `!` xD

Comment (antbern) [reply .github/workflows/deploy-swift.yml:168]: Alles gut 💯

**Review by lszetg — APPROVED** [review]
👏
